homeassistant/components/growatt_server/config_flow.py

```diff
@@ -0,0 +1,105 @@
+"""Config flow for growatt server integration."""
+import voluptuous as vol
+
+import growattServer
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME, CONF_USERNAME, CONF_PASSWORD
+from .const import DOMAIN, CONF_PLANT_ID  # noqa # pylint: disable=unused-import
+
+
+class GrowattServerConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow class."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialise growatt server flow."""
+        self.user_input = {}
+        self.api = growattServer.GrowattApi()
+        self.plants = {}
+        self.user_id = None
+
+    async def async_step_import(self, import_config):
+        """Import a config entry from configuration.yaml."""
+        if CONF_PLANT_ID in import_config and CONF_NAME in import_config:
+            return self.async_create_entry(
+                title=import_config[CONF_NAME], data=import_config
+            )
+
+        login_response = await self.hass.async_add_executor_job(
```

Comment:
Shouldn't we just pass the username and password of the imported config to the user step? We can store the plant id and name in an instance attribute that is available in the plant step.